File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    length = len(elements)
    result = []
    answer = 0
    
    for i in range(length):
        add = i + 1
        for j in range(length):
            end = j + add
            if end > length:
                result.append(sum(elements[j:] + elements[:end - length]))
            else:
                result.append(sum(elements[j:j + add]))
        

    return len(set(result))

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const length = elements.length
    const sequence = new Array(length ** 2).fill(0)

    
    for (let i = 0; i < length; i++) {
        sequence[i] = elements[i]
    }
    
    for (let i = length; i < sequence.length; i++) {
        const toAdd = (Math.floor(i / length) + i % length) % length
        sequence[i] = sequence[i - length] + elements[toAdd]
    }
    
    const answer = new Set(sequence).size
    return answer;
}